src/realm/group_shared.cpp

```diff
@@ -1860,6 +1859,8 @@ void SharedGroup::low_level_commit(uint_fast64_t new_version)
     out.set_versions(new_version, oldest_version);
     // Recursively write all changed arrays to end of file
     ref_type new_top_ref = out.write_group(); // Throws
+    m_free_space = out.get_free_space();
+    m_used_space = out.get_file_size() - m_free_space;
```

Comment:
What about only getting this info as needed when `get_stats()` is called? Otherwise we add a bit of performance overhead on every commit for finding something out that people may not use (`get_file_size` at least does an `fstat()` and `get_free_space()` must access every element in the free list).